docs/custom-images-contract.rst

```diff
@@ -0,0 +1,113 @@
+.. _custom-images-contract:
+
+======================
+Custom Images contract
+======================
+
+The contract is a set of requirements for crawler custom docker images to comply running with Scrapy Cloud.
+
+The contract is already implemented in Python and is known as :ref:`scrapinghub-entrypoint-scrapy <sh-entrypoint-scrapy>`. If you don’t use this package with your crawler image, you should implement Custom Images contract by yourself, or your image will not work properly in Scrapy Cloud.
+
+
+Contract statements
+-------------------
+
+1. Docker image should be able to run via ``start-crawl`` command without arguments.
+
+.. code-block:: bash
+
+    docker run myscrapyimage start-crawl
+
+2. Docker image should be able to return a spiders list via ``list-spiders`` command without arguments.
+
+.. code-block:: bash
+
+   docker run myscrapyimage list-spiders
+
+3. Crawler should be able to get all needed params using :ref:`system environment variables <environment-variables>`.
+
+
+.. _environment-variables:
+
+Environment variables
```

Comment:
All these variables are required, right? I think a quick intro paragraph to the section saying that they are "required because of this, this and that" would make this section more clear.